lib/buffer.js

```diff
@@ -588,9 +588,10 @@ function bidirectionalIndexOf(buffer, val, byteOffset, encoding, dir) {
   } else if (byteOffset < -0x80000000) {
     byteOffset = -0x80000000;
   }
-  byteOffset = +byteOffset;  // Coerce to Number.
+  // Coerce to Number. Values like null and [] become 0.
+  byteOffset = +byteOffset;
+  // If the offset is undefined, "foo", {}, coerces to NaN, search whole buffer.
   if (isNaN(byteOffset)) {
```

Comment:
If we're going to explicitly coerce the value first, then let's go ahead and use `Number.isNaN()` instead. Which explicitly checks for `NaN`, w/o first coercing the argument.